### ISSUE TITLE
Implement support for predict-time ratings in FunkSVD

### ISSUE THREAD

**Issue by ghost** [lenskit/lenskit#174]
(originally reported in Trac by @elehack on 2012-06-05 19:02:41)

Currently the FunkSVD rating predictor does not support prediction using ratings supplied at predict time; it can only use the user's ratings that were available at model-build time.

We want to modify the predictor to use the user's ratings at predict time.

The plan:
- Initialize the user feature vector. If the user has a user-feature vector in the model, use that as the initial vector; otherwise, use a vector composed of the average value over all users for each feature.
- Run the FunkSVD gradient descent algorithm over the user's ratings to train their user feature vector. Only the user feature update step, not the item feature update step, will be used. In addition, we may need to use a different regularization term. Depend on the standard regularization term; if we need separate model-build and predict regularization terms (or learning rates), we will use context-sensitive injection to configure them.

**Note**: This issue has been automatically migrated from Bitbucket
Created by [grouplens](https://bitbucket.org/grouplens) on 2013-02-01T22:07:57.550401+00:00, last updated: 2013-02-01T22:35:26.337641+00:00
